pipeline/load.py

```diff
@@ -0,0 +1,288 @@
+        logging.info("Error with provided artist name: %s", artist_name)
+        return -1
+
+    if type_id == -1:
+        logging.info("Error with provided release_type: %s", release_type)
+        return -1
+
+    try:
+        check_query = """SELECT release_id FROM release
+                        WHERE (release_name = %s
+                        AND release_date = %s
+                        AND artist_id = %s
+                        AND type_id = %s);"""
+
+        cursor.execute(check_query, (release_name,
+                       release_date, artist_id, type_id))
+        release_id = cursor.fetchone()
+
+        if release_id:
+            logging.info(
+                "'%s' already exists with release ID %s.", release_name, release_id[0])
+            return release_id[0]
+
+        insert_query = """INSERT INTO release (release_name, release_date, artist_id, type_id)
+                            VALUES (%s, %s, %s, %s)
+                            RETURNING release_id;"""
+        cursor.execute(insert_query, (release_name,
+                                      release_date, artist_id, type_id))
+        release_id = cursor.fetchone()
+        logging.info(
+            "Release '%s' of '%s' was added to the database.", release_id, release_name)
+        return release_id[0]
+
+    except psycopg2.Error:
+        logging.error("Error inserting release: %s", release_name)
+        return -1
+
+
+def insert_release_genres(release_id: int, genre_id: int, genre_name: str,
+                          release_name: str, cursor: psycopg2.extensions.cursor) -> None:
+    """
+    Inserts release genres into the database if it doesn't already exist.
+    Logs whether the release genre was added or already exists.
+    """
+    try:
+        check_query = """SELECT EXISTS
+                        (SELECT 1 FROM release_genre
+                        WHERE release_id = %s
+                        AND genre_id = %s);"""
+
+        cursor.execute(check_query, (release_id, genre_id))
+        exists = cursor.fetchone()[0]
+
+        if exists:
+            logging.info(
+                "Release '%s' already has '%s' as a genre.", release_name, genre_name)
+        else:
+            insert_query = """INSERT INTO release_genre (release_id, genre_id)
+                                VALUES (%s, %s);"""
+            cursor.execute(insert_query, (release_id, genre_id))
+            logging.info(
+                "'%s' was added to '%s' successfully.", genre_name, release_name)
+    except psycopg2.Error:
+        logging.error(
+            "Error adding '%s' to '%s' ID: %s", genre_name, release_name, release_id)
+
+
+def main_load(sales_df: pd.DataFrame) -> None:
+    """
+    Loads data from the sales dataframe into the database.
+    """
+    config_log()
+    connection = get_connection()
+    cursor = get_cursor(connection)
+
+    sales_df["genres"] = sales_df["genres"].apply(ast.literal_eval)
+
+    for _, row in sales_df.iterrows():
```

Comment:
Aren't we never meant to iterate over a Pandas DataFrame? Maybe this could be achieved by applying a function to the DataFrame?